testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/x509/X509DirectGrantTest.java

```diff
@@ -264,6 +267,8 @@ public void loginCertificateNotExpired() throws Exception {
 
     @Test
     public void loginCertificateExpired() throws Exception {
+        Assume.assumeFalse("Time offset is causing integer overflow. With the old store it works, because root authentication session has also timestamp overflown, this is not true for the new store so the test is failing.", keycloakUsingProviderWithId(AuthenticationSessionProvider.class, "map"));
+
         X509AuthenticatorConfigModel config =
                 new X509AuthenticatorConfigModel()
                     .setCertValidationEnabled(true)
```

Comment:
Would this test be re-enabled once this PR is ready, or would it be a separate issue?

Reply:
This is one of the things why I sent this just as a draft. Maybe I am missing something but it seems to enable this test we would need to make `Time.currentTime()` return long instead of int, which would require a lot of changes. That is why I decided to ignore this test for now and based on the discussion we have here, we will decide how to approach this.

To give you more details on why this is passing before my PR:
1. The test is moving in time 50 years to the future (after the year 2038 where integer timestamp overflows)
2. [This](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/managers/ClientSessionCode.java#L176) is the line where all of the magic happens :laughing:. `Time.currentTime()`  returns negative number. However, also timestamp is a negative number because we use `int` [here](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/authSession/MapRootAuthenticationSessionProvider.java#L102) (Not sure it is exactly the place, anyway it is one of the places where I changed `int` -> `long` in this PR).
3. Given this ^ the condition `lifespan > Time.currentTime() - timestamp` passes and the test works as expected. After my changes in this PR `timestamp` started to be positive number which breaks the condition.

That is why I think the only solution is to create some follow-up issue that would change `Time.currentTime` to long to make this test pass with the new store.

Reply:
@mhajas - please open a new issue for this. I'd assume one can move the time to 2037 (the year before 2038) to show the expiry of a certificate without relying on an overflow.

Reply:
I once gave it a start for #11053 and it turned out to be so big that I doubt that someone will be able review it in the near future (unfortunately). Therefore recommend to look at that new issue independently. With what you wrote about which certificate would be involved here, that analysis should be part of that other issue.